### ISSUE TITLE
aws-logging-cloudwatch: Emitted logs not being deleted from local database

### ISSUE THREAD

**Issue by hirengosalia86** [aws-amplify/amplify-android#2772]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin

### Amplify Categories

Not applicable

### Gradle script dependencies

<details>

```groovy
// Put output below this line
implementation("com.amplifyframework:core:2.15.0")
implementation("com.amplifyframework:aws-logging-cloudwatch:2.15.0")

```

</details>

### Environment information

<details>

```
------------------------------------------------------------
Gradle 7.6.1
------------------------------------------------------------

Build time:   2023-02-24 13:54:42 UTC
Revision:     3905fe8ac072bbd925c70ddbddddf4463341f4b4

Kotlin:       1.7.10
Groovy:       3.0.13
Ant:          Apache Ant(TM) version 1.10.11 compiled on July 10 2021
JVM:          11.0.22 (Amazon.com Inc. 11.0.22+7-LTS)
OS:           Mac OS X 14.3.1 x86_64




```

</details>

### Describe the bug

bulkDelete function called in syncLogEventsWithCloudwatch fails on deleteing, when there are more than 1 events. This causes log being duplicated multiple times in cloudwatch

### Code Snippet

```java
internal suspend fun bulkDelete(eventIds: List<Long>) = withContext(coroutineDispatcher) {
        contentUri
        val whereClause = "${LogEventTable.COLUMN_ID} in (?)" // number of '?' should match count of eventIds.
        database.delete(
            LogEventTable.TABLE_LOG_EVENT,
            whereClause,
            arrayOf(eventIds.joinToString(","))
        )
    }

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

**Comment by joon-won** [comment]
Hi @hirengosalia86, thank you for your report. Could you provide full log for this issue?

**Comment by hirengosalia86** [comment]
Hi @joon-won , there are no log outputs, but as you can see in the bulkDelete function, if there are more than 1 eventIds (1,2,3) there only exists 1 '?' corresponding to this, making it as a string '1,2,3'

**Comment by hirengosalia86** [comment]
@joon-won 

```
internal suspend fun bulkDelete(eventIds: List<Long>) = withContext(coroutineDispatcher) {
        contentUri
       val params =  List(eventIds.size, {"?"})
        val whereClause = "${LogEventTable.COLUMN_ID} in (${params.joinToString(",")})"
        database.delete(
            LogEventTable.TABLE_LOG_EVENT,
            whereClause,
            eventIds.toTypedArray()
        )
    }
```
This should fix it.

**Comment by joon-won** [comment]
Thank you for your reply and suggestion . However I do not believe the current code will fail when multiple ids are passed. `arrayOf()` function will make it Array<String> type `[1,2,3]`, and since this is a sqlite in its core, string representation should be working fine. If you were still noticing the deletion failing when you were passing multiple ids, someone in our team will take a look into the issue.

**Comment by hirengosalia86** [comment]
Agreed, string representation works fine in sqlite, but above will be arrayOf() on a string. Making it 
```
Delete from <table_name> where <cloumn> in ('1,2,3')
```

**Comment by joon-won** [comment]
Thank you for providing your example and explanation, I see the point. I will share this with the team, and we will look into the issue

**Comment by joon-won** [comment]
This issue is resolved with Amplify Android `2.16.1` please let us know if the issue persists
